paddle/fluid/framework/new_executor/instruction/cinn_jit_instruction.cc

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/framework/new_executor/instruction/cinn_jit_instruction.h"
+
+#include "paddle/cinn/hlir/dialect/jit_kernel_op.h"
+#include "paddle/cinn/hlir/dialect/runtime_dialect.h"
+#include "paddle/cinn/hlir/framework/instruction.h"
+#include "paddle/fluid/framework/paddle2cinn/transform_type.h"
+
+namespace paddle {
+namespace framework {
+
+// TODO(Aurelius84): Think deeply what's the responsibility is it.
+// Currently it assumes CinnLaunchContext role.
+class JitContext {
+ public:
+  cinn_buffer_t* GetCinnBufferOfVar(const std::string& name) {
+    auto res = paddle2argument_.find(name);
+    PADDLE_ENFORCE_NE(
+        res,
+        paddle2argument_.end(),
+        platform::errors::NotFound(
+            "Variable(%s) not found in compilation result", name));
+    return static_cast<cinn_buffer_t*>(res->second);
+  }
+
+  // NOTE(Aurelius84): Before running each instruction, we should share Tensor
+  // memory from paddle scope with cinn_buffer_t from cinn scope including
+  // inputs and outputs.
+  void ShareMemToCinn(const std::string& var_name,
+                      const phi::Place& place,
+                      Scope* scope) {
+    cinn_buffer_t* buffer = GetCinnBufferOfVar(var_name);
+    auto* tensor = scope->GetVar(var_name)->GetMutable<phi::DenseTensor>();
+    // TODO(Aurelius84): Maybe we should consider to unify the Scope
+    // structure between paddle and cinn, so that we don't need to develop
+    // the glue code.
+    buffer->memory = reinterpret_cast<uint8_t*>(tensor->mutable_data(
```

Comment:
Remove use of `mutable_data` in the future

Reply:
Got it， I will remove it next PR